### ISSUE TITLE
Database migration UNIQUE constraint failed: irc_channel.network_id, irc_channel.name (PTP)

### ISSUE THREAD

**Issue by ohjk** [autobrr/autobrr#1896]
**Version**
1.56

**Describe the bug**
Received this error first time running autobrr after upgrading from 1.55 to 1.56.  I can roll back to 1.55 and it still runs fine

Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","time":"2024-12-26T07:13:56Z","message":"maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined"}
Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:56Z","message":"Beginning database schema upgrade from version 66 to version: 72"}
Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:56Z","message":"Database integrity check.."}
Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:56Z","message":"Database integrity check OK!"}
Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:56Z","message":"Creating database backup: /home/jk/.config/autobrr/autobrr.db_sv66_2024-12-26.07-13-56.backup"}
Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:56Z","message":"Database backup created at: /home/jk/.config/autobrr/autobrr.db_sv66_2024-12-26.07-13-56.backup"}
Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:56Z","message":"Upgrading Database schema to version: 67"}
Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:56Z","message":"Upgrading Database schema to version: 68"}
Dec 26 07:13:56 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:56Z","message":"Upgrading Database schema to version: 69"}
Dec 26 07:13:57 circa autobrr[2889929]: {"level":"info","module":"database","type":"sqlite","time":"2024-12-26T07:13:57Z","message":"Upgrading Database schema to version: 70"}
Dec 26 07:13:57 circa autobrr[2889929]: {"level":"fatal","module":"database","type":"sqlite","stack":[{"func":"(*DB).migrateSQLite","line":"148","source":"sqlite.go"},{"func":"(*DB).openSQLite","line":"83","source":"sqlite.go"},{"func":"(*DB).Open","line":"78","source>

**Comment by zottelbeyer** [comment]
Had this issue as well. It was due to manually creating a config for the IRC channel #ptp-announce.
You can manually fix it by removing the channel from autobrr.db table "irc_channel"

**Comment by zze0s** [comment]
That log doesn't contain why it failed unfortunately so hard to tell exactly what went wrong here.

> Had this issue as well. It was due to manually creating a config for the IRC channel #ptp-announce. You can manually fix it by removing the channel from autobrr.db table "irc_channel"

I don't think it should fail from the migration to update that, but could be wrong 🤔

**Comment by kthrow** [comment]
I ran into the same issue. Due to using `restart: unless-stopped` this ended up producing about 80GB of database backups over the course of 8 hours or so until I started getting disk usage warning emails.

I ended up solving temporarily by downgrading back to 1.55.0 and deleting all backups with `_sv66` in them aside from the first one temporarily.

**Comment by zze0s** [comment]
Sorry about that! Do you have any logs that could help?

The thing that generates all those backups is most likely the integrity check which is only done on SQLite. I'll implement a limit for it tomorrow.

There were so many db changes in this update and I tested it pretty extensively and other maintainers tested on their setups as well but some stuff still seem to be semi broken.

**Comment by kthrow** [comment]
I think this is the relevant fatal error:

```

{"level":"fatal","module":"database","type":"sqlite","stack":[{"func":"(*DB).migrateSQLite","line":"148","source":"sqlite.go"},{"func":"(*DB).openSQLite","line":"83","source":"sqlite.go"},{"func":"(*DB).Open","line":"78","source":"database.go"},{"func":"main","line":"98","source":"main.go"},{"func":"main","line":"272","source":"proc.go"},{"func":"goexit","line":"1700","source":"asm_amd64.s"}],"error":"failed to execute migration #69: constraint failed: UNIQUE constraint failed: irc_channel.network_id, irc_channel.name (2067)","time":"2024-12-26T16:49:43-05:00","message":"could not migrate db"}
```

**Comment by ohjk** [comment]
> Had this issue as well. It was due to manually creating a config for the IRC channel #ptp-announce. You can manually fix it by removing the channel from autobrr.db table "irc_channel"

in 1.55 I removed the #ptp-announce channel I had manually added when #ptp-announce-dev stopped working, then stopped the service, unpacked the 1.56 release and restarted it and its working perfectly.  

@zze0s looks like the upgrade automatically cleaned out all the old logs

all I have now in ~/.config/autobrr/logs is: autobrr-2024-12-22T17-26-54.378.log  autobrr.log

**Comment by zze0s** [comment]
@ohjk Hm ok, will debug tomorrow with an older install with that channel changed and run upgrades on it and see if it could be the cause of issues.

@kthrow I wonder if something might not have been cleaned properly in the past potentially. Could you join our Discord (link on the docs in the header) and see if you can use something like https://sqlitebrowser.org/ to view the db and check what you have in the table `irc_channel` and DM that to me?

**Comment by kthrow** [comment]
Whoops, I ended up solving the problem as above and now I don't have a backup of the bad database state 🙃 

The fix was essentially as the OP described. My PTP entry in v1.55.0 had 2 channels listed, where the `-dev` variant appeared as disabled. Not sure if it was always like that or entered that state after the migration failed (given 67 and 68 had no errors, I assume the database was left on version 68 and not 66?). I removed the non-suffixed version and upgraded and it went fine.

I'm guessing this is because of the migration that sets the channel name to the non-suffixed version if it is the `-dev` version, with left me with 2 channels with the same name on the same network, which isn't allowed.
